### PR TITLE
ServiceBus SDK changes to support RBAC changes

### PR DESCRIPTION
**Pull request by bainian12345**
- Removed dependency on ADAL
- Rename ManagedServiceIdentity TokenProvider to ManagedIdentityProvider
- Added connection string support for Managed Identity tokenProvider
- Replaced signatures for AAD TokenProvider with user defined callback interface 
- Added corresponding tests

### CODE REVIEW

**Comment by mssfang** [comment]
Make sure there is no SpotBugs and CheckStyle errors. The CI will not pass until there is no more errors.
Current changes produces 1 SpotBugs error and 37 CheckStyle errors and 4 CheckStyle warnings after I ran maven. 

"mvn -Dgpg.skip -DskipTests -Dinclude-non-shipping-modules -Ddata-plane install -f servicebus\data-plane\pom.xml"

"mvn -Dgpg.skip -DskipTests -Dinclude-non-shipping-modules -Ddata-plane site -f servicebus\data-plane\pom.xml"

I just ran the

**Comment by bainian12345** [comment]
Should be free of these error now.

---
In reply to: [501008301](https://github.com/Azure/azure-sdk-for-java/pull/3876#issuecomment-501008301) [](ancestors = 501008301)

File: servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/ConnectionStringBuilder.java [hunk 1]
```diff
@@ -354,7 +382,13 @@ public String toString() {
                 connectionStringBuilder.append(String.format(Locale.US, "%s%s%s%s", KEY_VALUE_PAIR_DELIMITER, TRANSPORT_TYPE_CONFIG_NAME,
                         KEY_VALUE_SEPARATOR, this.transportType.toString()));
             }
-
+            
+            if (this.authentication != null)
+            {
```
Comment (mssfang) [inline servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/ConnectionStringBuilder.java]: This should be a CheckStyle warning. No left curly brace for a new line.

**Comment by mssfang** [comment]
LGTM

**Pull request merged**